### PR TITLE
feat(llm): add gptme provider for managed service

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
## Summary

Adds the built-in `gptme` provider for the gptme cloud managed LLM proxy service, with full RFC 8628 Device Flow authentication.

### Provider
- Provider name: `gptme` (use with `gptme -m gptme/claude-sonnet-4-6`)
- OpenAI-compatible — reuses existing `chat_openai`/`stream_openai`
- Pass-through model support (models proxied to upstream providers)

### Auth CLI
- `gptme-auth login` — Device Flow auth with `--no-browser` flag
- `gptme-auth logout` — Remove stored credentials
- `gptme-auth status` — Show login state and expiry
- URL-hash-based token storage for multi-instance support

### Configuration
- `GPTME_CLOUD_API_KEY` env var as fallback
- `GPTME_CLOUD_BASE_URL` for custom endpoints
- Default: `https://fleet.gptme.ai/v1`

**Usage:**
```bash
gptme-auth login                        # authenticate via Device Flow
gptme -m gptme/claude-sonnet-4-6        # use proxied model
```

**Supersedes** #1525 (auth CLI consolidated here).

## Test plan
- [x] 17 unit tests passing (provider registration, token lifecycle, auth priority, URL normalization, expires_at edge cases)
- [x] mypy clean
- [x] ruff clean

### CODE REVIEW

**Comment by TimeToBuildBob** [comment]
@greptileai review

**Comment by ErikBjare** [comment]
Conflicts with https://github.com/gptme/gptme/pull/1525

Keep one of them and move the remaining stuff into the the one that remains.

**Comment by ErikBjare** [comment]
Also I don't think we should call it the "gptme-ai" provider, it should just be the "gptme" provider (implicitly the managed service aka gptme-cloud).

**Comment by ErikBjare** [comment]
Similarly `GPTME_AI_API_KEY` should probably be `GPTME_CLOUD_API_KEY` and something similar for `GPTME_AI_BASE_URL` into `GPTME_CLOUD_BASE_URL`

Also `Base URL priority: token's server_url > GPTME_AI_BASE_URL env var > https://api.gptme.ai/v1` is wrong, it should be `Base URL priority: token's server_url > GPTME_CLOUD_BASE_URL env var > https://fleet.gptme.ai/v1`

**Comment by TimeToBuildBob** [comment]
## Addressed All Review Feedback

**Commit**: 280d8a0f6

### Erik's feedback:
- **Renamed `gptme-ai` → `gptme`** as the provider name (all files, all references)
- **Env vars renamed**: `GPTME_AI_API_KEY` → `GPTME_CLOUD_API_KEY`, `GPTME_AI_BASE_URL` → `GPTME_CLOUD_BASE_URL`
- **Default base URL fixed**: `api.gptme.ai` → `fleet.gptme.ai/v1`
- **Consolidated with PR #1525**: `gptme-auth login/logout/status` replaces `gptme-auth gptme-ai`

### Greptile fixes:
- `expires_at=0` now correctly treated as expired (was bypassing check)
- Env var URL gets `/v1` suffix normalization (consistent with token URL)
- Added `gptme` to warning suppression list in `models.py`

### New features from #1525 consolidation:
- `gptme-auth login` — Device Flow with `--no-browser` flag, progress dots, auto-browser-open
- `gptme-auth logout` — remove stored credentials
- `gptme-auth status` — show login state and expiry
- URL-hash-based token storage for multi-instance support (`gptme-cloud-<hash>.json`)

**Tests**: 17 passing (3 new: `expires_at=0`, URL hash, env var normalization)

**Supersedes PR #1525** — all auth CLI code is now here.

**Comment by TimeToBuildBob** [comment]
Done — consolidated #1525 into this PR. All auth CLI code (login/logout/status) is now here. #1525 can be closed.

File: gptme/llm/llm_gptme_ai.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    if token_data and token_data.get("server_url"):
+        server_url = token_data["server_url"].rstrip("/")
+        if not server_url.endswith("/v1"):
+            server_url += "/v1"
+        return server_url
+
+    # Check env var
+    env_url = config.get_env("GPTME_AI_BASE_URL")
+    if env_url:
+        return env_url.rstrip("/")
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply gptme/llm/llm_gptme_ai.py]: Fixed in 280d8a0f6 — env URL now gets `/v1` suffix normalization.

File: gptme/llm/llm_gptme_ai.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+    try:
+        data = json.loads(token_path.read_text())
+    except (json.JSONDecodeError, OSError) as e:
+        logger.warning(f"Failed to read gptme-ai token: {e}")
+        return None
+
+    # Check expiration (with 60s buffer)
+    expires_at = data.get("expires_at", 0)
+    if expires_at and time.time() > expires_at - 60:
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply gptme/llm/llm_gptme_ai.py]: Fixed in 280d8a0f6 — `expires_at=0` is now correctly treated as expired. Added test `test_load_token_zero_expires_at`.

**Comment by TimeToBuildBob** [comment]
@greptileai review

File: gptme/llm/llm_gptme.py [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+        if error == "authorization_pending":
+            continue
+        if error == "slow_down":
+            interval = min(interval + 5, 30)
+            continue
+        raise RuntimeError(f"Device flow failed: {error}")
+
+    raise RuntimeError("Device flow timed out — code expired")
+
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply gptme/llm/llm_gptme.py:224]: Fixed in 4a72e3430 — removed the 91-line `device_flow_authenticate` function. The Device Flow implementation lives in `auth.py` (`auth_login` command). `_save_token` stays as it's imported and used by `auth.py`.

**Comment by TimeToBuildBob** [comment]
## Greptile dead code fixed

**Commit**: 4a72e3430

Removed the 91-line `device_flow_authenticate` function from `llm_gptme.py` — it was never called (the Device Flow implementation lives in `auth.py`'s `auth_login` command). `_save_token` stays as it's legitimately used by `auth.py`.

All CI checks were passing before this change (typecheck, lint, tests). Replied to Greptile's inline thread.
